419: WP GraphQL Call To Action Data

rpeterman-gp opened the pull request:
Closes #419 

- fetch CTA data with WP GraphQL API
- fix styling of banner and load-under CTA messages
- move bio segments list pagination into sidebar list component for performance and reusability
- update categories subcategory list to use paginated sidebar list
- add search to sidebar list component

## To Review

- [x] Checkout and update lando `main` branch.
- [x] Import latest backup of live WP env into lando.
- [x] Add one of each type of Call To Action to Site CTA Regions. Include the Planet Hip Hop playlist CTA.
- [x] Checkout Branch.
- [x] Run `yarn`.
- [x] Run `yarn dev:start`.
- [x] Go to [localhost:3000](https://localhost:3000).

> ...then...

- [ ] Ensure all CTA Regions show the expected Call To Action.
- [ ] Dismiss the banner CTA.
- [x] Go to http://localhost:3000/categories/arts-culture-media/music/planet-hip-hop
- [x] Ensure Banner and Load-under regions update to the targeted CTA.
- [x] Ensure playlist CTA is shown in the expected inline region.
- [x] Dismiss banner CTA and refresh page.
- [x] Ensure next expected CTA is shown. If the banner and load-under were the same CTA, the load under CTA should have changes as well.
- [x] Repeat dismiss and refresh steps, and ensure each type of CTA's styles look good.
- [ ] Browse around stories, episodes, programs, etc. Ensure CTA are shown as expected.
- [x] Go to http://localhost:3000/categories/conflict-justice
- [x] Test pagination and search of the Subcategories sidebar list.
- [x] Go to http://localhost:3000/contributors/daniel-ofman
- [x] Test pagination and search of the segments sidebar list.

rpeterman-gp committed:
fix(alias): type errors

rpeterman-gp committed:
fix(bio): segment pagination wrapping

rpeterman-gp committed:
fix(filterctamessages): category targeting

- remove end padding from info cta content

rpeterman-gp committed:
Merge pull request #413 from PRX/fix/cta-filtering-category-target

HOTFIX: CTA category targeting

rpeterman-gp committed:
fix(term): state type generic

rpeterman-gp committed:
chore: trigger amplify build

rpeterman-gp committed:
feat(newsletter): wp gql data fetch

- update comments of various files for accuracy

rpeterman-gp committed:
fix([...alias]): newsletter component data prop

rpeterman-gp committed:
feat(cta): wp gql data

- move app data to redux store

rpeterman-gp committed:
Merge branch 'production' into feat/419-wp-gql-cta-data

rpeterman-gp committed:
feat(sidebarlist): add pagination and search

- category fetch gets all children
- update bio and category to use sidebarlist changes

rpeterman-gp committed:
fix(ctas): app cta msg styles

rpeterman-gp committed:
fix(bio): follow links keys

- balance sidebar header text wrap

brandonhundt reviewed (approved):
@rpeterman-gp this is great! Nice work. Love the search filter you snuck in!

One thing, featured story pages aren't rendering the inline-end CTA region. (or it is set up wrong?) 
![Screenshot 2023-09-01 at 1 26 41 PM](https://github.com/PRX/theworld.org/assets/369739/e67a3178-e818-43df-892f-74a786339210)
![Screenshot 2023-09-01 at 1 28 11 PM](https://github.com/PRX/theworld.org/assets/369739/b45151d3-3933-4e04-97cf-b4866bf9709d)

rpeterman-gp committed:
feat(story): update cta regions

- program ctas
- episode ctas
- term ctas
- category ctas
- update htmlcontent cleanhtml to remove line returns

rpeterman-gp committed:
fix(appplayer): layout of controls on mobile

brandonhundt reviewed (approved):
@rpeterman-gp :shipit: !!! Thanks!

rpeterman-gp committed:
fix(build): type errors
